configuration/src/main/java/io/airlift/configuration/SlatedForRemoval.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.airlift.configuration;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.time.format.DateTimeFormatter;
+import java.util.Locale;
+
+/**
+ * Annotated configuration setting is slated for removal.
+ * The annotated property name is suffixed with {@value #PROPERTY_SUFFIX}
+ * and removal date in {@link #PROPERTY_DATE_FORMAT}.
+ * Must be on same method as {@code @Config}.
+ */
+@Documented
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.METHOD)
+public @interface SlatedForRemoval
+{
+    DateTimeFormatter VALUE_DATE_FORMAT = DateTimeFormatter.ofPattern("MMM yyyy", Locale.US);
+    String PROPERTY_SUFFIX = ".slated-for-removal-";
+    DateTimeFormatter PROPERTY_DATE_FORMAT = DateTimeFormatter.ofPattern("yyyy-MM", Locale.US);
+
+    /**
+     * Describes date until which config is guaranteed to be retained.
+     * <p>
+     * Must be in the format accepted by {@link #VALUE_DATE_FORMAT}, for example {@code "Oct 2019"}.
```

Comment:
I would use the ISO format for months (YYYY-MM), too. It's unambiguous and consistent with how we treat dates/times everywhere else.

Reply:
I thought this way will be more readable.